core/src/main/java/io/camunda/migrator/RuntimeMigrator.java

```diff
@@ -205,10 +279,23 @@ public Map<String, ActInstance> getActiveActivityIdsById(ActivityInstance activi
     return activeActivities;
   }
 
-  record ActInstance(String activityId, String subProcessInstanceId) {
+  public void setBatchSize(int batchSize) {
+    this.batchSize = batchSize;
+  }
+
+  record ActInstance(String id, String subProcessInstanceId) {
   }
 
   public void setRetryMode(boolean retryMode) {
     this.retryMode = retryMode;
   }
+
+  protected <T> void paginate(Long maxCount, Function<Integer, Set<T>> query, Consumer<T> callback) {
+    for (int offset = 0; offset < maxCount; offset = offset + batchSize) {
+      System.out.println("Max count: " + maxCount + ", offset: " + offset + ", batch size: " + batchSize + ""); // TODO
```

Comment:
💭 THis is not great since there's no info to which query we list the page.

Reply:
Yes. Not sure right now how to improve this.